modules/git/ref.go

```diff
@@ -24,17 +36,17 @@ func (ref *Reference) ShortName() string {
 	if ref == nil {
 		return ""
 	}
-	if strings.HasPrefix(ref.Name, "refs/heads/") {
-		return ref.Name[11:]
+	if strings.HasPrefix(ref.Name, BranchPrefix) {
+		return ref.Name[branchLen:]
 	}
-	if strings.HasPrefix(ref.Name, "refs/tags/") {
-		return ref.Name[10:]
+	if strings.HasPrefix(ref.Name, TagPrefix) {
+		return ref.Name[tagLen:]
 	}
-	if strings.HasPrefix(ref.Name, "refs/remotes/") {
-		return ref.Name[13:]
+	if strings.HasPrefix(ref.Name, RemotePrefix) {
+		return ref.Name[remoteLen:]
```

Comment:
Maybe we can do `strings.TrimPrefix` (still fast) then we do not need these `xxxLen`

Reply:
Minor nitpick on my side, TrimPrefix includes on itself HasPrefix again.

Reply:
So I pushed the TrimPrefix change

@gusted, I am not sure to understand what you mean here.

Do you suggest we do TrimPrefix directly without checking with HasPrefix because that's the same operation more or less, so it might be faster to not do twice ?

Reply:
> I am not sure to understand what you mean here.

The problem is that TrimPrefix itself again includes `HasPrefix` and some other (useless) checks in this scenario.

Reply:
Do you mean something like

 `func util.TrimPrefixIfExists(s string) (result string, trimmed bool)` 

```go
if result, trimmed = TrimPrefixIfExists(ref.Name); trimmed {
    return result
}
```
?